quantum/process_keycode/process_space_cadet.c

```diff
@@ -0,0 +1,142 @@
+/* Copyright 2019 Jack Humbert
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "process_space_cadet.h"
+
+#ifndef TAPPING_TERM
+  #define TAPPING_TERM 200
+#endif
+
+// ********** OBSOLETE DEFINES, STOP USING! (pls?) **********
+// Shift / paren setup
+#ifndef LSPO_KEY
+  #define LSPO_KEY KC_9
+#endif
+#ifndef RSPC_KEY
+  #define RSPC_KEY KC_0
+#endif
+
+// Shift / Enter setup
+#ifndef SFTENT_KEY
+  #define SFTENT_KEY KC_ENT
+#endif
+
+#ifdef DISABLE_SPACE_CADET_MODIFIER
+  #ifndef LSPO_MOD
+    #define LSPO_MOD KC_TRNS
+  #endif
+  #ifndef RSPC_MOD
+    #define RSPC_MOD KC_TRNS
+  #endif
+#else
+  #ifndef LSPO_MOD
+    #define LSPO_MOD KC_LSFT
+  #endif
+  #ifndef RSPC_MOD
+    #define RSPC_MOD KC_RSFT
+  #endif
+#endif
+// **********************************************************
+
+// Shift / paren setup
+#ifndef LSPO_KEYS
+  #define LSPO_KEYS KC_LSFT, LSPO_MOD, LSPO_KEY
+#endif
+#ifndef RSPC_KEYS
+  #define RSPC_KEYS KC_RSFT, RSPC_MOD, RSPC_KEY
+#endif
+
+// Control / paren setup
+#ifndef LCPO_KEYS
+  #define LCPO_KEYS KC_LCTL, KC_LCTL, KC_9
+#endif
+#ifndef RCPO_KEYS
+  #define RCPO_KEYS KC_RCTL, KC_RCTL, KC_0
+#endif
+
+// Alt / paren setup
+#ifndef LAPO_KEYS
+  #define LAPO_KEYS KC_LALT, KC_LALT, KC_9
+#endif
+#ifndef RAPO_KEYS
+  #define RAPO_KEYS KC_RALT, KC_RALT, KC_0
+#endif
+
+// Shift / Enter setup
+#ifndef SFTENT_KEYS
+  #define SFTENT_KEYS KC_RSFT, KC_TRNS, SFTENT_KEY
```

Comment:
Should these `KC_TRNS` actually be `KC_NO`? Is there even a use case for using regular keycodes instead of just `MOD_`s here?

Reply:
It doesn't matter what a user shoves in the second value just so long as `IS_MOD` can handle it and return true or false correctly. So a user could pass in KC_A, KC_NO, 0, etc.

Reply:
Oh, so it does check whether it's a mod, sorry, should have read the code 😆

My point still stands though - by only accepting `MOD_*` for the first two parameters you *know* they're supposed to be modifiers. *And*... you can OR them together!

Reply:
Ah I see, there is precedence today for them being keycodes as that is what Space Cadet Shift and Shift Enter use today. It would probably be fine to change them as a quick scan of the source shows no keymap has defined them differently today. However keymaps not in qmk could be doing so which would break them with that change.

Reply:
@skullydazed thoughts?

Reply:
I think I agree with @fauxpark here, that `MOD_*` is the better choice. Even if there's no technical different between the two `MOD_` conveys intent better.

Reply:
sounds good, will change it

Reply:
@skullydazed does `MOD_` still report keycode down / up for that modifier to the pc?

Reply:
Not sure, but there is `register_mod` and `unregister_mod`, which I think it does.

Reply:
oh right, derp, this cold is getting to me

Reply:
fixed